_pytest/warnings.py

```diff
@@ -72,8 +72,8 @@ def catch_warnings_for_item(item):
             unicode_warning = False
 
             if compat._PY2 and any(isinstance(m, compat.UNICODE_TYPES) for m in warn_msg.args):
-                new_args = [compat.safe_str(m) for m in warn_msg.args]
-                unicode_warning = warn_msg.args != new_args
+                new_args = [m.encode('ascii', 'replace') for m in warn_msg.args]
```

Comment:
i fear this one could count as a breaking change as output characters are removed now

Reply:
You are absolutely right, thanks. Rebased.